File: 백준/Bronze/9093. 단어 뒤집기/단어 뒤집기.py
n = int(input())

for i in range(n):
    arr = input()
    pos = 0
    stack = []
    while(1):
        if(pos == len(arr) or arr[pos] == ' '):
            for j in range(len(stack) - 1, -1, -1):
                print(stack[j], end="")
            stack = []
            pos += 1
            print(" ", end="")
        if(pos - 1 == len(arr)): break
        stack.append(arr[pos])
        pos += 1
    print()

File: 백준/Bronze/9093. 단어 뒤집기/단어 뒤집기.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(void) {
	int n;
	scanf("%d", &n);
	getchar();
	char arr[1001];
	while (n--) {
		scanf("%[^\n]", arr);

		char stack[1001];
		int pos = 0, i = 0, len = strlen(arr);
		while (1) {
			if (arr[pos] == ' ' || pos == len) {
				for (int j = i - 1; j >= 0; j--) {
					printf("%c", stack[j]);
				}
				i = 0, pos++;
				printf(" ");
			}
			if (pos - 1 == len) break;
			stack[i++] = arr[pos++];
		}
		printf("\n");
		getchar();
	}
	
	return 0;
}